src/cursor/common.rs

```diff
@@ -104,7 +131,7 @@ where
                     }
                     return Poll::Ready(Some(Ok(doc)));
                 }
-                None if !self.exhausted => {
+                None if !self.exhausted && !self.pinned_connection.is_invalid() => {
```

Comment:
this condition needs to be added a few lines above too. Alternatively, we can just remove that call to `self.start_get_more()` in the branch where a doc is found, since it doesn't actually do anything until it gets polled.

Reply:
Took the second approach - I think it makes the flow a little easier to follow.